demo/parser_demo.cpp

```diff
@@ -129,14 +129,61 @@ class PrintVisitor : public ast::Visitor {
             depth--;
         }
     };
-    void visit(ast::Expression* node) override{
-
+    void visit(ast::Expression* node) override {
+        if (node == nullptr) {
+            fmt::print("{:|>{}}- [Expression]> null\n", "", depth);
+        } else {
+            // if binary
+            if (node->operand2) {
+
+                fmt::print("{:|>{}}- [BinaryOperator]> {}\n", "", depth,
```

Comment:
This `print` can be factored out of the if/else

Reply:
Why? I think that with this visitor we want to distinguish binary and unary operations. Off course, we can simply print "[Operator] >". What do you think? Is it better?

Reply:
I thought they were both exactly the same